### PR TITLE
Upgrade Go version to 1.21

### PR DESCRIPTION
**Pull request by gapra-msft**
Merge after 10.24 release

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -76,4 +75,4 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-go 1.19
+go 1.21
```
Comment (adreed-msft) [inline go.mod:78]: I'm curious: Any particular reason to *not* use Go 1.22?

File: main.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ import (
 var glcm = common.GetLifecycleMgr()
 
 func main() {
-	rand.Seed(time.Now().UnixNano()) // make sure our random numbers actually are random (but remember, use crypto/rand for anything where strong/reliable randomness is required
+	rand.New(rand.NewSource(time.Now().UnixNano())) // make sure our random numbers actually are random (but remember, use crypto/rand for anything where strong/reliable randomness is required
```
Comment (adreed-msft) [inline main.go:39]: This is incorrect. `rand.New` returns a new `*rand.Rand`, and does not set the seed globally. Please track down all uses of `rand` and ensure they use this `*rand.Rand`

**Review by adreed-msft — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (adreed-msft) [reply main.go:39]: Actually lemme double correct this-- According to the docs, as of Go 1.20 this is no longer necessary at all. https://pkg.go.dev/math/rand#Seed

Comment (gapra-msft) [reply main.go:39]: will remove this line entirely

File: go.mod (hunk 1 above)
Comment (gapra-msft) [reply go.mod:78]: linting doesnt support 1.22 yet so I just set this to 121 for now. we can upgrade to 1.22 if we want to.

Comment (gapra-msft) [reply go.mod:78]: actually it wasnt linting, it was the trivy vulnerability scanner that doesnt support 1.22 yet

**Review by pranavmalik-msft — APPROVED** [review]

File: main.go [hunk 3]
```diff
@@ -21,23 +21,18 @@
...
-
-	"github.com/Azure/azure-storage-azcopy/v10/cmd"
-	"github.com/Azure/azure-storage-azcopy/v10/common"
 )
 
 // get the lifecycle manager to print messages
 var glcm = common.GetLifecycleMgr()
 
 func main() {
-	rand.Seed(time.Now().UnixNano()) // make sure our random numbers actually are random (but remember, use crypto/rand for anything where strong/reliable randomness is required
```
Comment (siminsavani-msft) [inline main.go:39]: What is the reasoning behind removing this randomization? /gen

**Review by siminsavani-msft — APPROVED** [review]

File: main.go (hunk 3 above)
Comment (gapra-msft) [reply main.go:39]: This is no longer necessary with the latest Go version. Now rand is by default generating a random seed.

**Review by siminsavani-msft — APPROVED** [review]

**Pull request merged**